inventory_management_system_api/services/catalogue_category_property.py

```diff
@@ -115,3 +120,112 @@ def create(
+                        "values"
+                    )
+
+    def update(
+        self,
+        catalogue_category_id: str,
+        catalogue_item_property_id: str,
+        catalogue_item_property: CatalogueItemPropertyPatchRequestSchema,
+    ) -> CatalogueItemPropertyOut:
+        """
+        Update a property at the catalogue category level by its id
+
+        Property changes will be propagated down through the catalogue items and items when required where there are
+        children
+
+        :param catalogue_category_id: The ID of the catalogue category to update
+        :param catalogue_item_property_id: The ID of the catalogue item property within the category to update
+        :param catalogue_item_property: The property values to update
+        :raises MissingRecordError: If the catalogue category doesn't exist, or the catalogue item property doesn't
+                                    exist within the specified catalogue category
+        """
+
+        update_data = catalogue_item_property.model_dump(exclude_unset=True)
+
+        # Obtain the existing catalogue category to validate against
+        stored_catalogue_category = self._catalogue_category_repository.get(catalogue_category_id)
+        if not stored_catalogue_category:
+            raise MissingRecordError(f"No catalogue category found with ID: {catalogue_category_id}")
+
+        # Attempt to locate the property
+        existing_property_out: CatalogueItemPropertyOut = None
+        for prop in stored_catalogue_category.catalogue_item_properties:
+            if prop.id == catalogue_item_property_id:
+                existing_property_out = prop
+                break
+
+        if not existing_property_out:
+            raise MissingRecordError(f"No catalogue item property found with ID: {catalogue_item_property_id}")
+
+        # Modify the name if necessary and check it doesn't cause a conflict
+        updating_name = "name" in update_data and update_data["name"] != existing_property_out.name
+        if updating_name:
+            existing_property_out.name = update_data["name"]
+            utils.check_duplicate_catalogue_item_property_names(stored_catalogue_category.catalogue_item_properties)
+
+        if "allowed_values" in update_data:
+            self._check_valid_allowed_values_update(
+                existing_property_out.allowed_values, catalogue_item_property.allowed_values
+            )
+
+        CatalogueCategoryPostRequestPropertySchema.check_valid_allowed_values(
+            catalogue_item_property.allowed_values, existing_property_out.model_dump()
+        )
+
+        catalogue_item_property_in = CatalogueItemPropertyIn(**{**existing_property_out.model_dump(), **update_data})
+
+        # Run all subsequent edits within a transaction to ensure they will all succeed or fail together
+        with mongodb_client.start_session() as session:
+            with session.start_transaction():
```

Comment:
Do we need to handle any exceptions here as well as all other places where we are using sessions/transactions?

Reply:
No I don't believe so, the only errors I think it would likely give are those we might get in any database edit e.g. a connection error which will currently all give 500 errors anyway as we don't catch them anywhere. I can't see anything specific to just sessions & transactions.